fe/fe-core/src/main/java/com/starrocks/planner/FragmentNormalizer.java

```diff
@@ -578,11 +579,16 @@ List<Expr> getPartitionRangePredicates(List<Expr> conjuncts,
             if (range.isEmpty()) {
                 continue;
             }
-            range = toClosedOpenRange(range);
+            Optional<Range> optRange = Optional.empty();
+            try {
+                optRange = Optional.ofNullable(toClosedOpenRange(range));
+            } catch (Throwable ignored) {
+            }
+
             Pair<Long, Range<PartitionKey>> partitionKeyRange = rangeMap.get(i);
             // when the range is to total cover this partition, we also cache it
-            if (!range.isEmpty()) {
-                selectedRangeMap.put(partitionKeyRange.first, range.toString());
+            if (optRange.isPresent() && !optRange.get().isEmpty()) {
```

Comment:
Is it possiable two different partition has the same selectedRange?

Reply:
Impossible, at present, only single column range partition is take into consideration, so partitions are not intersected with each other. A selected range is just a sub-range of a certain partition.